elixir/apps/domain/lib/domain/auth/adapters/openid_connect.ex

```diff
@@ -85,6 +85,22 @@ defmodule Domain.Auth.Adapters.OpenIDConnect do
     end
   end
 
+  @impl true
+  def sign_out(%Provider{} = provider, %Identity{} = identity, redirect_url) do
+    config = config_for_provider(provider)
+
+    case OpenIDConnect.end_session_uri(config, %{
+           id_token_hint: identity.provider_state["id_token"],
+           post_logout_redirect_uri: redirect_url
+         }) do
+      {:ok, end_session_uri} ->
+        {:ok, identity, end_session_uri}
+
+      {:error, _reason} ->
```

Comment:
Just curious -- if this fails for whatever reason, would it make sense to clear our session cookie here as well? They'd be auto-logged in again whenever they click the "Sign in" button via the IdP, but this would prevent the flow where a user clicks "sign out", then gets autoredirected back to the Dashboard because their Firezone session is actually still valid.

Reply:
We will reset the cookie in those cases, it happens in auth controller. Moreover we reset cookie even if you don't have an active session.

Reply:
Ah thanks for clarifying.